controlpanel/api/cluster.py

```diff
@@ -218,230 +110,68 @@ def url(self):
         return f"https://{ingresses[0].spec.rules[0].host}"
 
 
-class IAMRole:
-
-    def __init__(self, role_name):
-        self.policy = {
-            "Version": "2012-10-17",
-            "Statement": [
-                {
-                    "Effect": "Allow",
-                    "Principal": {
-                        "Service": "ec2.amazonaws.com",
-                    },
-                    "Action": "sts:AssumeRole",
-                },
-                {
-                    "Effect": "Allow",
-                    "Principal": {
-                        "AWS": iam_arn(f"role/{settings.K8S_WORKER_ROLE_NAME}"),
-                    },
-                    "Action": "sts:AssumeRole",
-                },
-            ],
-        }
-        self.role_name = role_name
-
-    def save(self):
-        try:
-            aws.create_role(self.role_name, self.policy)
-        except Exception as e:
-            if not is_ignored_exception(e):
-                raise e
-
-    def allow_assume_role_with_saml(self):
-        self.policy["Statement"].append(dict(SAML_STATEMENT))
-
-    def allow_assume_role_with_oidc(self, sub):
-        stmt = dict(OIDC_STATEMENT)
-        stmt["Condition"] = {
-            "StringEquals": {
-                f"{settings.OIDC_DOMAIN}/:sub": sub,
-            }
-        }
-        self.policy["Statement"].append(stmt)
-
-
-class S3AccessPolicy(dict):
-    POLICY_NAME = "s3-access"
+class Bucket:
```

Comment:
=> `class S3Bucket` for consistency with pretty much everything else ☺️

Reply:
The name of `S3Bucket` is a mistake that I am slowly correcting

Reply:
Just kidding, I'll change the name

Reply:
😱 You know you freaked out for a moment 😝